staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -176,13 +176,94 @@ type Histogram struct {
 	*dto.Histogram
 }
 
-// GetHistogramFromGatherer collects a metric from a gatherer implementing k8s.io/component-base/metrics.Gatherer interface.
+// HistogramVec wraps a slice of Histogram.
+// Note that each Histogram must have the same number of buckets.
+type HistogramVec []*Histogram
+
+// GetSampleCount aggregates the sample count of each inner Historgram.
+func (vec HistogramVec) GetSampleCount() uint64 {
+	var count uint64
+	for _, hist := range vec {
+		count += hist.GetSampleCount()
+	}
+	return count
+}
+
+// GetSampleSum aggregates the sample sum of each inner Historgram.
+func (vec HistogramVec) GetSampleSum() float64 {
+	var sum float64
+	for _, hist := range vec {
+		sum += hist.GetSampleSum()
+	}
+	return sum
+}
+
+// Quantile first aggregates inner buckets of each Histogram, and then
+// computes q-th quantile of a cumulative histogram.
+func (vec HistogramVec) Quantile(q float64) float64 {
+	var buckets []bucket
+
+	for _, hist := range vec {
+		for i, bckt := range hist.Bucket {
+			if len(buckets) <= i {
```

Comment:
so let's say the first histogram has buckets different from the second histogram such that the second hist's first bucket is smaller than the first hist's first bucket. then the second histogram's quantisation will be incorrectly counted in the cumulative buckets right?

or do we know that all Histograms in a HistogramVc have the same buckets? if so, I think checking for `i == 0` is easier to understand

Reply:
We assume all Histograms in a HistogramVc have the same buckets.

> I think checking for i == 0 is easier to understand

Updated.